clock.go

```diff
@@ -36,3 +37,7 @@ var _systemClock Clock = systemClock{}
 func (systemClock) Now() time.Time {
 	return time.Now()
 }
+
```

Comment:
Do these exported names not need comments?

Reply:
We don't need comments here as the `systemClock` is not being exported, it is just a private variable satisfying an interface. 

Most likely I will move this systemClock to zapcore and exported it, if that happens I will add the proper comments for the exported type.

Reply:
Ah, I see. Thanks for the clarification!